proxy-robots-module/src/main/java/orwell/proxy/robot/RobotInputSetVisitor.java

```diff
@@ -0,0 +1,79 @@
+package orwell.proxy.robot;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import orwell.messages.Controller;
+
+
+/**
+ * Created by Michaël Ludmann on 5/12/15.
+ */
+public class RobotInputSetVisitor implements IRobotInputVisitor {
```

Comment:
There is a hidden time dependency between the different visit methods as if the one accepting an IRobot is not called last, it will not find all objects set.
One of the benefits of the visitor patter should be to be able to extend what can be done with some elements without modifying those objects but here you had to modify InputFire and InputMove so the behaviour is split between the visitor and the visited.